### ISSUE TITLE
ENH: show an warning when np.maximum received more than two inputs

### ISSUE THREAD

**Issue by xuwinnie** [numpy/numpy#27639]
### Proposed new feature or change:

it would be good to show an warning when np.maximum received more than two inputs, otherwise ppl(like me) may think it supports multiple inputs, and spend a lot of time trying to figure out whats wrong

**Comment by mattip** [comment]
Agreed, the error message is misleading.
```
>>> import numpy as np
>>> np.maximum(3, 4, 5)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: return arrays must be of ArrayType
>>> np.maximum(3, 4)
np.int64(4)
```

Its signature, like all ufuncs, is 
```
maximum(x1, x2, /, out=None, *, where=True, casting='same_kind', order='K', dtype=None, subok=True[, signature])
```

Changing the ufunc signature to be kwd-only would be a large backward incompatible change. Maybe we could do it via a deprecation process.

**Comment by seberg** [comment]
I would be up for slowly deprecating not using `out=` for ufuncs in general.  Could potentially escalate to a `VisibleDeprecationWarning` relatively soon, to make it easier for users to spot.

I would not be surprised if part of the reason for it not being kwarg-only, is that the speed difference between keyword and non-keyword use, used to be much bigger.

The actual change required is very contained, but in C (and of course it needs tests and tracking down the `DEPRECATE` pattern used elsewhere).

**Comment by mhvk** [comment]
I agree we should encourage keyword `out`, though I worry it is a lot of churn for not *that* much benefit. I guess the alternative is to just guard functions where people are easily misled -- `minimum` and `maximum` do seem good examples of that. A final alternative is to simply make the error message more explicit.

**Comment by seberg** [comment]
Yeah, I think when I wrote it, I didn't think too much about how much code may actually be passing `out` as arguments.  I can imagine a special case for minimum/maximum as they are more likely to be incorrectly used.
I also don't mind aiming for deprecation, but we should actually spend a bit of time to see if there are many users and then probably take it slow.

I honestly still like our old philosophy of just refusing to type undesired patterns.  Also because that is an additional way to nudge users away from it.  If we wish we would deprecate it one day (even if we may hesitate to do it), then not typing it feels very reasonable (maybe with docs to have at least some basis for that choice...).
